pkg/webui/console/store/middleware/logics/gateways.js

```diff
@@ -12,23 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { createLogic } from 'redux-logic'
+
+import sharedMessages from '../../../../lib/shared-messages'
 import api from '../../../api'
 import * as gateways from '../../actions/gateways'
+import { gsConfigSelector } from '../../../../lib/selectors/env'
+import { selectSelectedGateway } from '../../selectors/gateways'
+import createEventsConnectLogics from './events'
 import createRequestLogic from './lib'
 
+const getGatewayLogic = createRequestLogic({
+  type: gateways.GET_GTW,
+  async process ({ action }) {
+    const { payload, meta } = action
+    const { id = {}} = payload
+    const selector = meta.selector || ''
+    return api.gateway.get(id, selector)
+  },
+})
+
+const updateGatewayLogic = createRequestLogic({
+  type: gateways.UPDATE_GTW,
+  async process ({ action }) {
+    const { payload: { gatewayId, patch }} = action
+    const result = await api.gateway.update(gatewayId, patch)
+
+    return { ...patch, ...result }
+  },
+}, gateways.updateGatewaySuccess)
```

Comment:
```suggestion
})
```
Not necessary with the current implementation.